Go tests fail with Go 1.25/NixOS 26.05: TestDurationDecoder error type mismatch and flaky shell duration tests

While building `resticprofile-0.31.0` on NixOS 26.05, the build fails during the Go test phase. I originally hit this via the NixOS/nixpkgs package build, but the failing tests appear to be upstream test-suite issues.

There seem to be at least two separate problems:

1. `TestDurationDecoder` expects the exact old error type/string from `time.ParseDuration`, but the current Go toolchain returns `*time.parseDurationError`.
2. Some shell duration tests appear timing-sensitive/flaky in a sandboxed build environment.

## Environment

```text
resticprofile: 0.31.0
OS: NixOS 26.05
Build context: nixpkgs package build
Command: nixos-rebuild switch --upgrade --show-trace
```

## Failure 1: TestDurationDecoder error type mismatch

On one build attempt, the `shell` tests passed, but `util/maybe` failed:

```text
--- FAIL: TestDurationDecoder (0.00s)
    --- FAIL: TestDurationDecoder/5 (0.00s)
        duration_test.go:70:
            	Error Trace:	/build/source/util/maybe/duration_test.go:70
            	Error:      	Not equal:
            	            	expected: *errors.errorString(&errors.errorString{s:"time: invalid duration \"ns\""})
            	            	actual  : *time.parseDurationError(&time.parseDurationError{message:"invalid duration", value:"ns"})
            	Test:       	TestDurationDecoder/5
    --- FAIL: TestDurationDecoder/6 (0.00s)
        duration_test.go:70:
            	Error Trace:	/build/source/util/maybe/duration_test.go:70
            	Error:      	Not equal:
            	            	expected: *errors.errorString(&errors.errorString{s:"time: invalid duration \"invalid\""})
            	            	actual  : *time.parseDurationError(&time.parseDurationError{message:"invalid duration", value:"invalid"})
            	Test:       	TestDurationDecoder/6
FAIL
FAIL	github.com/creativeprojects/resticprofile/util/maybe	0.271s
```

This looks like the test is asserting the exact concrete error type/value returned by `time.ParseDuration`. With the current Go toolchain, invalid durations such as `ns` and `invalid` return `*time.parseDurationError` instead of the older `*errors.errorString`.

It may be more robust to assert that an error occurs and/or compare the semantic error message, rather than checking the exact concrete error type.

## Failure 2: flaky/timing-sensitive shell tests

On another build attempt, the failure was different and happened in the `shell` package:

```text
--- FAIL: TestSummaryDurationSignalledCommand (1.52s)
    command_test.go:480:
        Error: Max difference ... allowed is 500ms, but difference was 522.654542ms
    command_test.go:482:
        Error: "1522" is not less than "1500"

--- FAIL: TestSummaryDurationCommand (1.51s)
    command_test.go:455:
        Error: Max difference ... allowed is 500ms, but difference was 513.497777ms
    command_test.go:457:
        Error: "1513" is not less than "1500"

--- FAIL: TestInterruptShellCommand (3.88s)
    command_test.go:391:
        Error: An error is expected but got nil.

FAIL
FAIL    github.com/creativeprojects/resticprofile/shell    32.817s
```

The duration tests look timing-sensitive. In a sandboxed or busy build environment, the measured values can exceed the hard-coded threshold slightly, for example 1513 ms or 1522 ms where the test expects less than 1500 ms.

## Other build/check output

The build/check also includes OS-specific packages that have no Go files on Linux:

```text
github.com/creativeprojects/resticprofile/darwin: build constraints exclude all Go files
github.com/creativeprojects/resticprofile/schtasks: build constraints exclude all Go files
```

I am not sure if this part is caused by the upstream test command, the nixpkgs builder, or both, but I am including it because it appears in the same failed check phase.

## Related downstream issue

There is also a downstream nixpkgs build failure issue for this package:

https://github.com/NixOS/nixpkgs/issues/502783

## Current local workaround

In nixpkgs/NixOS, disabling checks allows the package build to continue:

```nix
nixpkgs.overlays = [
  (final: prev: {
    resticprofile = prev.resticprofile.overrideAttrs (old: {
      doCheck = false;
    });
  })
];
```

That is only a workaround. The likely upstream fixes are to make the duration decoder test less dependent on the exact concrete Go error type, and to make the shell timing tests less flaky in slower/sandboxed environments.

--- a/shell/command_test.go
+++ b/shell/command_test.go
@@ -434,9 +434,9 @@ func TestSummaryDurationCommand(t *testing.T) {
 	require.NoError(t, err)
 
 	// make sure the command ran properly
-	assert.WithinDuration(t, time.Now(), start.Add(1*time.Second), 500*time.Millisecond)
+	assert.WithinDuration(t, time.Now(), start.Add(1*time.Second), 4*time.Second)
 	assert.GreaterOrEqual(t, summary.Duration.Milliseconds(), int64(1000))
-	assert.Less(t, summary.Duration.Milliseconds(), int64(1500))
+	assert.Less(t, summary.Duration.Milliseconds(), int64(5000))
 }
 
 func TestSummaryDurationSignalledCommand(t *testing.T) {
@@ -459,9 +459,9 @@ func TestSummaryDurationSignalledCommand(t *testing.T) {
 	require.NoError(t, err)
 
 	// make sure the command ran properly
-	assert.WithinDuration(t, time.Now(), start.Add(1*time.Second), 500*time.Millisecond)
+	assert.WithinDuration(t, time.Now(), start.Add(1*time.Second), 4*time.Second)
 	assert.GreaterOrEqual(t, summary.Duration.Milliseconds(), int64(1000))
-	assert.Less(t, summary.Duration.Milliseconds(), int64(1500))
+	assert.Less(t, summary.Duration.Milliseconds(), int64(5000))
 }
 
 func TestStderr(t *testing.T) {

--- a/shell/command_unix_test.go
+++ b/shell/command_unix_test.go
@@ -21,17 +21,17 @@ func TestInterruptShellCommand(t *testing.T) {
 	cmd := NewSignalledCommand(mockBinary, []string{"test", "--sleep", "3000"}, sigChan)
 	cmd.Stdout = buffer
 
-	// Will ask us to stop in 100ms
+	// Will ask us to stop in 500ms
 	go func() {
-		time.Sleep(100 * time.Millisecond)
+		time.Sleep(500 * time.Millisecond)
 		sigChan <- syscall.SIGINT
 	}()
 	start := time.Now()
 	_, _, err := cmd.Run()
 	require.Error(t, err)
 
-	// check it ran for more than 100ms (but less than a second - the build agent can be slow)
+	// check it ran for more than 500ms (but well under the 3000ms sleep - the build agent can be slow)
 	duration := time.Since(start)
-	assert.GreaterOrEqual(t, duration.Milliseconds(), int64(100))
-	assert.Less(t, duration.Milliseconds(), int64(1000))
+	assert.GreaterOrEqual(t, duration.Milliseconds(), int64(500))
+	assert.Less(t, duration.Milliseconds(), int64(3000))
 }

